src/coreclr/tests/src/JIT/opt/AssertionPropagation/ArrBoundBinaryOp.cs

```diff
@@ -0,0 +1,1376 @@
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    private static int returnCode = 100;
+
+    public static int Main(string[] args)
+    {
+        int[] testIndices = Enumerable.Range(-50, 50)
+                // also add some corner case values
+                .Concat(new[] { int.MinValue, int.MinValue + 1, int.MaxValue, int.MaxValue - 1 })
+                .ToArray();
+
+        foreach (int index in testIndices)
+        {
+            CompareActions(index, Tests.Test000_cns, Tests.Test000_cns);
```

Comment:
You are passing the same actions to `CompareActions`. Did you intend
`CompareActions(index, Tests.Test000_cns, Tests.Test000_var)` ?
Same for all other calls to `CompareAction`.

Reply:
Oops, indeed! thanks